src/ext/framework/mod.rs

```diff
@@ -254,90 +257,103 @@ impl Framework {
-                            message.content[position + built.len()..]
-                                .split_whitespace()
-                                .map(|arg| arg.to_owned())
-                                .collect::<Vec<String>>()
-                        };
-
-                        if let Some(x) = command.min_args {
-                            if args.len() < x as usize {
-                                return;
+                            let args = if command.use_quotes {
+                                utils::parse_quotes(&message.content[position + built.len()..])
+                            } else {
+                                message.content[position + built.len()..]
+                                    .split_whitespace()
+                                    .map(|arg| arg.to_owned())
+                                    .collect::<Vec<String>>()
+                            };
+
+                            if let Some(x) = command.min_args {
+                                if args.len() < x as usize {
+                                    return;
+                                }
                             }
-                        }
 
-                        if let Some(x) = command.max_args {
-                            if args.len() > x as usize {
-                                return;
+                            if let Some(x) = command.max_args {
+                                if args.len() > x as usize {
+                                    return;
+                                }
                             }
-                        }
 
-                        if !command.required_permissions.is_empty() {
-                            let mut permissions_fulfilled = false;
+                            if !command.required_permissions.is_empty() {
+                                let mut permissions_fulfilled = false;
 
-                            if let Some(member) = message.get_member() {
-                                let cache = CACHE.read().unwrap();
+                                if let Some(member) = message.get_member() {
+                                    let cache = CACHE.read().unwrap();
 
-                                if let Ok(guild_id) = member.find_guild() {
-                                    if let Some(guild) = cache.get_guild(guild_id) {
-                                        let perms = guild.permissions_for(message.channel_id, message.author.id);
+                                    if let Ok(guild_id) = member.find_guild() {
+                                        if let Some(guild) = cache.get_guild(guild_id) {
+                                            let perms = guild.permissions_for(message.channel_id, message.author.id);
 
-                                        permissions_fulfilled = perms.contains(command.required_permissions);
+                                            permissions_fulfilled = perms.contains(command.required_permissions);
```

Comment:
Side-note: the indentation here has gotten a little insane.

Note-to-self: revise this later